sling/task/learner.cc

```diff
@@ -56,7 +57,16 @@ void LearnerTask::Train(Task *task, myelin::Network *model) {
     }
 
     // Run evaluation.
-    Evaluate(epoch_, model);
+    if (!Evaluate(epoch_, model)) {
+      done_ = true;
+      break;
+    }
+
+    // Checkpoint model.
+    if (epoch_ >= last_checkpoint_ + checkpoint_interval_) {
+      Checkpoint(epoch_, model);
+      last_checkpoint_ += checkpoint_interval_;
```

Comment:
The epoch_ variable is an atomic counter, which can be updated while running, so I think it is safer to do it this way to avoid drift in the checkpoint intervals.